### PR TITLE
Fix adding unnecessary colons on Mapped Types

### LINKED ISSUE #7174: Format error syntax with language typescript
BEFORE RUN FORMAT
`type A = { readonly [key in keyof B] };`
AFTER RUN FORMAT
`type A = { readonly [key in keyof B]:  };`

### PR DESCRIPTION
**Pull request by cola119**
fixes #7174
This is my first time submitting PR. :bow:

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by armano2** [comment]
@cola119 can you add test case?
```
type Anys = { [K in Keys]+? };
```

File: tests/typescript/conformance/types/mappedType/mappedType.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 type Keys = 'option1' | 'option2';
 type Flags = { [K in Keys]: boolean };
-
+type Anys = { [K in Keys] };
```
Comment (sosukesuzuki) [inline tests/typescript/conformance/types/mappedType/mappedType.ts]: I'm not familiar with the background, but I think it is better to create new directory named `/tests/typescript_mapped_type` than adding tests to `/tests/typescripts/conformance`.
(I was previously pointed on a similar thing from @thron0. See https://github.com/prettier/prettier/pull/6604#issuecomment-538934227)

Comment (cola119) [reply tests/typescript/conformance/types/mappedType/mappedType.ts]: It makes sense. I'll try it.

Comment (cola119) [reply tests/typescript/conformance/types/mappedType/mappedType.ts]: wait, should I create new tests at `/tests/typescript_mapped_type`  while leaving `/tests/typescripts/conformance/mappedType.ts`? Or move `/tests/typescripts/conformance/mappedType.ts` to `/tests/typescript_mapped_type`?

Comment (cola119) [reply tests/typescript/conformance/types/mappedType/mappedType.ts]: I understood the problem by reading the above conversation so I'm going to create new dir  while keeping `/conformance/mappedType.ts` before :bow:

File: tests/typescript_mapped_type/mapped-type.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+type Keys = 'option1' | 'option2';
+type A = { [K in Keys] };
+type B = { [K in Keys]+? };
```
Comment (cola119) [inline tests/typescript_mapped_type/mapped-type.ts]: @armano2 is this as intended? 🤔

Comment (armano2) [reply tests/typescript_mapped_type/mapped-type.ts]: its valid syntax

File: changelog_unreleased/typescript/pr-7174.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#### Fix adding unnecessary colons on Mapped Types ([#7221](https://github.com/prettier/prettier/pull/7221) by [@cola119](https://github.com/cola119))
```
Comment (thorn0) [inline changelog_unreleased/typescript/pr-7174.md]:
```suggestion
#### Fix printing of mapped types with the template type omitted ([#7221](https://github.com/prettier/prettier/pull/7221) by [@cola119](https://github.com/cola119))
```

**Pull request merged**